server/app/views/v1/stacks/_stack.json.jbuilder

```diff
@@ -15,5 +15,17 @@ json.variables latest_rev.variables
 json.services stack.grid_services.to_a do |grid_service|
   json.partial! 'app/views/v1/grid_services/grid_service', grid_service: grid_service
 end
-json.parent stack.has_parent? ? { name: stack.parent_name } : nil
-json.children stack.children.pluck(:name).map { |child_name| { name: child_name } }
+
+json.parent do
+  if stack.has_parent?
+    json.id stack.parent.to_path
+    json.name stack.parent.name
+  end
+end
+
+json.children do
+  stack.children.map do |child_stack|
```

Comment:
You can't use `json.*` from within `map`, it's a DSL function with side-effects, this is results in either the `children` being completely missing from the JSON response, or a single JSON object like `{"children":{"id":"development/deptest-foo","name":"deptest-foo"}}`.

Look at how the jbuilder thing for arrays works for e.g. the similar `GridService.secrets` case: https://github.com/kontena/kontena/blob/11e97b7c6803510ae71c750807697beaa7ca8867/server/app/views/v1/grid_secrets/show.json.jbuilder#L5-L8